pkg/planner/core/point_get_plan.go

```diff
@@ -2213,19 +2352,46 @@ func tryDeletePointPlan(ctx base.PlanContext, delStmt *ast.DeleteStmt, resolveCt
 		if ctx.GetSessionVars().TxnCtx.IsPessimistic {
 			pointGet.Lock, pointGet.LockWaitTime = getLockWaitTime(ctx, &ast.SelectLockInfo{LockType: ast.SelectLockForUpdate})
 		}
-		return buildPointDeletePlan(ctx, pointGet, pointGet.dbName, pointGet.TblInfo, delStmt.IgnoreErr)
-	}
-	if batchPointGet := tryWhereIn2BatchPointGet(ctx, selStmt, resolveCtx); batchPointGet != nil {
+		pointPlan, dbName, tbl, colsInWhereClause = pointGet, pointGet.dbName, pointGet.TblInfo, pointGet.colsInWhereClause
+	} else if batchPointGet := tryWhereIn2BatchPointGet(ctx, selStmt, resolveCtx); batchPointGet != nil {
 		if ctx.GetSessionVars().TxnCtx.IsPessimistic {
 			batchPointGet.Lock, batchPointGet.LockWaitTime = getLockWaitTime(ctx, &ast.SelectLockInfo{LockType: ast.SelectLockForUpdate})
 		}
-		return buildPointDeletePlan(ctx, batchPointGet, batchPointGet.dbName, batchPointGet.TblInfo, delStmt.IgnoreErr)
+		pointPlan, dbName, tbl, colsInWhereClause = batchPointGet, batchPointGet.dbName, batchPointGet.TblInfo, batchPointGet.colsInWhereClause
 	}
+	if pointPlan != nil {
+		// We don't check the output names, since they are expended to all columns of the table.
```

Comment:
Does this mean that the tryDeletePointPlan command does not require permission checks for returned columns?

Reply:
Yes, here the SelectStmt is constructed manually. In `buildSchemaFromFields`, the output names will be all the columns in the table, which we don't need to check.